src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -597,6 +606,91 @@ private ProcessWaitState GetWaitState()
             return _waitStateHolder._state;
         }
 
+        private static (uint userId, uint groupId) GetUserAndGroupIds(ProcessStartInfo startInfo)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(startInfo.UserName));
+
+            (uint? userId, uint? groupId) = GetUserAndGroupIds(startInfo.UserName);
+
+            if (userId == null || groupId == null)
+            {
+                throw new Win32Exception(SR.Format(SR.UserDoesNotExist, startInfo.UserName));
+            }
+
+            return (userId.Value, groupId.Value);
+        }
+
+        private unsafe static (uint? userId, uint? groupId) GetUserAndGroupIds(string userName)
+        {
+            // First try with a buffer that should suffice for 99% of cases.
+            Interop.Sys.Passwd? passwd;
+            const int BufLen = 1024;
```

Comment:
consider  defining as 0 in DEBUG to force exercising the fallback path. This is a common trick.

Reply:
Done.

Reply:
Obviously it can't be `0`, since we double in the fallback path.  But I'll start it at `2` in DEBUG cases, which is not large enough to hold a normal user name.